staging/src/k8s.io/api/apps/v1/types.go

```diff
@@ -173,6 +207,15 @@ type StatefulSetSpec struct {
 	// consists of all revisions not represented by a currently applied
 	// StatefulSetSpec version. The default value is 10.
 	RevisionHistoryLimit *int32 `json:"revisionHistoryLimit,omitempty" protobuf:"varint,8,opt,name=revisionHistoryLimit"`
+
+	// persistentVolumeClaimPolicy describes the lifecycle of persistent volume claims
+	// created from volumeClaimTemplates. By default, all persistent volume claims are
+	// created as needed and retained until manually deleted. This policy allows the
+	// lifecycle to be altered, for example by deleting persistent volume claims when
+	// their stateful set is deleted, or when their pod is scaled down. This requires the
+	// StatefulSetAutoDeletePVC feature gate to be enabled, which is alpha.
+	// +optional
+	PersistentVolumeClaimPolicy *StatefulSetPersistentVolumeClaimPolicy `json:"persistentVolumeClaimPolicy,omitempty" protobuf:"bytes,9,opt,name=persistentVolumeClaimPolicy"`
```

Comment:
I think it would be better to be more specific `PersistentVolumeClaimRetentionPolicy` as discussed in the other thread, and name the struct `StatefulSetPersistentVolumeClaimRetentionPolicy`.  That roughly matches `RevisionHistoryLimit` in breadth which is managing retention of history (i.e. we didn't need to create a generic policy for revisions, we simply tried to capture the policy with just enough future proofing).

Reply:
Done